### PR TITLE
Extension supports new OpenAPI generator

### PR DESCRIPTION
**Pull request by bensonnw**

### CODE REVIEW

File: src/util/GenerateProject.ts [hunk 1]
```diff
@@ -58,83 +60,100 @@ export async function generateProject(fileUri): Promise<void> {
...
-                // remove .openapi-generator created by generator
-                util.deleteFile(path.resolve(srcDir, '.openapi-generator-ignore'));
-
-            } catch (err) {
-                console.error(err);
-            }
-        }
-    });
+			}
+		);
```
Comment (ericglau) [inline src/util/GenerateProject.ts]: Change tabs to spaces.

File: src/util/GenerateProject.ts [hunk 2]
```diff
@@ -58,83 +60,100 @@ export async function generateProject(fileUri): Promise<void> {
...
-                ncp(generatedPath, srcDir, function (err) {
-                    if (err) {
-                        return console.error(err);
-                    }
-                    console.log('Moved ' + generatedPath + ' to ' + srcDir);
+        vscode.window.withProgress( // run the command with a "progress" dialog
+			{
+				location: vscode.ProgressLocation.Notification,
+				title: "Generating the MicroProfile Rest Client interface template..."
+			}, async () => {
```
Comment (ericglau) [inline src/util/GenerateProject.ts]: The spacing looks odd here, should change the tabs to spaces.

**Review by ericglau — APPROVED** [review]
Tabs should be spaces.  The rest looks good to me.

File: src/util/GenerateProject.ts (hunk 1 above)
Comment (bensonnw) [reply src/util/GenerateProject.ts]: Changed to spaces

File: src/util/GenerateProject.ts (hunk 2 above)
Comment (bensonnw) [reply src/util/GenerateProject.ts]: Changed to spaces

File: src/util/util.ts [hunk 3]
```diff
@@ -36,11 +36,28 @@ export async function specifyTargetFolder(uri: vscode.Uri): Promise<vscode.Uri>
...
+export async function getPackageName(srcDir: string, pathSep: string): Promise<string> {
+    // looking for "/java/" from the path
+    var index = srcDir.toLowerCase().indexOf(pathSep + "java" + pathSep);
+    let defaultPackageName: string;
+    if (index > -1) {
+        index = index + 6;
+        defaultPackageName = srcDir.substring(index).replace(pathSep, "."); // use everything after the "/java/" to be the package name
+    }
+    
+    var exp = /^[a-z][a-z0-9_]*(\.[a-z0-9_]+)*$/;   // validate the package name
```
Comment (kathrynkodama) [inline src/util/util.ts:48]: When validating the package name, the default name does not replace all of the `/` with `.` . The user is then unable to submit the auto-filled package name until they manually replace the `/` with `.`
image

Comment (bensonnw) [reply src/util/util.ts:48]: Fixed the "replace" issue, please give it a try.

**Review by ericglau — APPROVED** [review]

**Review by kathrynkodama — APPROVED** [review]

**Pull request merged**